### DISCUSSION TITLE
Formula contains filename CELL("filename")

### DISCUSSION THREAD

**Discussion by mrl22** [PHPOffice/PhpSpreadsheet#3804] [Q&A] [answered]
Hello

I have been asked to work on a project incorporating PhpSpreadsheet to read an existing Excel spreadsheet.

My concern is that one of the cells has a formula calculated based on the filename of the Excel sheet using `CELL("filename")`.

Here is the full formula for context, but don't put any time into reading it, it looks a mess :) 
```
=LEFT(MID(CELL("filename"),FIND("[",CELL("filename"))+1, FIND("]",CELL("filename"))-FIND("[",CELL("filename"))-1),FIND(".",MID(CELL("filename"),FIND("[",CELL("filename"))+1, FIND("]",CELL("filename"))-FIND("[",CELL("filename"))-1))-1)
```

I'd like to know if anyone has experience with this and can tell me if I will run into issues with PhpSpreadsheet using this.

Trying to scope out what needs to be done ahead of time and if I need to take the uploaded filename and produce this myself.

I have tried to search for answers on this but the GitHub search and google seems to see it as "cell filename" which is useless.

Thank you so much for your attention and participation.

**Comment by oleibman** [comment, marked as answer]
PhpSpreadsheet does not support calculating the value of the CELL function. However, it will read and write files that use the function, so, as long as your program does not need to be aware of the calculated value, you should be okay.

**Comment by mrl22** [comment]
Thank you @oleibman; I assumed this after reading over the docs > calculation engine pages.

I will need to use the PHP upload filename instead, as I need some information from it. 

I am only looping the data and inserting into MySQL, so nothing major.
